aim/agent/aid/universes/aci/tenant.py

```diff
@@ -436,41 +595,29 @@ def _push_aim_resources(self):
-                                    dn += '/tag-%s' % self.tag_name
-                                    tags.append({"tagInst__%s" % obj.keys()[0]:
-                                                 {"attributes": {"dn": dn}}})
-                        LOG.debug("Pushing %s into APIC: %s" %
-                                  (method, to_push + tags))
-                        # Multiple objects could result from a conversion, push
-                        # them in a single transaction
                         try:
-                            if method == base_universe.DELETE:
-                                for obj in to_push + tags:
+                            if method == base_universe.CREATE:
+                                # Set ownership before pushing the request
+                                to_push = self.ownership_mgr.set_ownership_key(
+                                    to_push)
+                                LOG.debug("POSTING into APIC: %s" % to_push)
+                                self._post_with_transaction(to_push)
+                            else:
+                                to_delete, to_update = (
+                                    self.ownership_mgr.set_ownership_change(
+                                        to_push))
+                                LOG.debug("DELETING from APIC: %s" % to_delete)
+                                for obj in to_delete:
                                     attr = obj.values()[0]['attributes']
                                     self.aci_session.DELETE(
                                         '/mo/%s.json' % attr.pop('dn'))
-                            else:
-                                with self.aci_session.transaction(
-                                        top_send=True) as trs:
-                                    for obj in to_push + tags:
-                                        attr = obj.values()[0]['attributes']
-                                        mo, parents_rns = decompose(
-                                            attr.pop('dn'), obj.keys()[0])
-                                        rns = dn_mgr.filter_rns(parents_rns)
-                                        getattr(getattr(self.aci_session, mo),
-                                                method)(*rns, transaction=trs,
-                                                        **attr)
-                                # Object creation was successful, change object
-                                # state
+                                LOG.debug("UPDATING in APIC: %s" % to_update)
+                                # Update object ownership
+                                self._post_with_transaction(to_update,
+                                                            modified=True)
+                            # Object creation was successful, change object
+                            # state
+                            if method == base_universe.CREATE:
```

Comment:
this line is redundant of line#599. So we should move self.creation_succeeded(aim_object) up to around line#605.